core/src/main/java/com/uber/m3/tally/ScopeImpl.java

```diff
@@ -62,41 +62,43 @@ class ScopeImpl implements Scope, TestScope {
         this.separator = builder.separator;
         this.tags = builder.tags;
         this.defaultBuckets = builder.defaultBuckets;
+        this.clock = builder.clock;
     }
 
     @Override
     public Counter counter(String name) {
         return counters.computeIfAbsent(name, ignored ->
-                // NOTE: This will called at most once
-                new CounterImpl(this, fullyQualifiedName(name))
+            // NOTE: This will be called at most once
+            new CounterImpl(this, fullyQualifiedName(name))
         );
     }
 
     @Override
     public Gauge gauge(String name) {
         return gauges.computeIfAbsent(name, ignored ->
-                // NOTE: This will called at most once
-                new GaugeImpl(this, fullyQualifiedName(name)));
+            // NOTE: This will be called at most once
+            new GaugeImpl(this, fullyQualifiedName(name)));
     }
 
     @Override
     public Timer timer(String name) {
         // Timers report directly to the {@code StatsReporter}, and therefore not added to reporting queue
         // i.e. they are not buffered
-        return timers.computeIfAbsent(name, ignored -> new TimerImpl(fullyQualifiedName(name), tags, reporter));
+        return timers.computeIfAbsent(name, ignored -> new TimerImpl(clock, fullyQualifiedName(name), tags, reporter));
     }
 
     @Override
     public Histogram histogram(String name, @Nullable Buckets buckets) {
         return histograms.computeIfAbsent(name, ignored ->
-                // NOTE: This will be called at most once
-                new HistogramImpl(
-                        this,
-                        fullyQualifiedName(name),
-                        tags,
-                        Optional.ofNullable(buckets)
-                                .orElse(defaultBuckets)
-                )
+            // NOTE: This will be called at most once
+            new HistogramImpl(
+                clock,
+                this,
+                fullyQualifiedName(name),
+                tags,
+                Optional.ofNullable(buckets)
```

Comment:
maybe the following? It will reduce overheads of Optional

`buckets != null ? buckets : defaultBuckets`